core/gui/src/app/dashboard/component/user/user-dataset/user-dataset-explorer/dataset-detail.component.html

```diff
@@ -224,6 +241,41 @@ <h6 style="font-weight: lighter; font-size: 0.9em">Choose a Version:</h6>
               Version Size: {{ formatSize(currentDatasetVersionSize) }}
             </div>
           </div>
+
+          <!-- Simple README creation in sidebar -->
+          <div *ngIf="userDatasetAccessLevel === 'WRITE' && !hasReadmeFile()" class="create-readme-section">
+            <button
+              nz-button
+              nzType="dashed"
+              nzBlock
+              nzSize="small"
+              (click)="showReadmeForm = !showReadmeForm">
+              <i nz-icon nzType="file-add"></i>
+              {{ showReadmeForm ? 'Cancel' : 'Create README.md' }}
+            </button>
+
+            <!-- README creation form -->
+            <div *ngIf="showReadmeForm">
```

Comment:
you should try to reuse the file editor component, a separate form seems to be redundant